api-gateway/src/main/resources/static/scripts/visits/visits.controller.js

```diff
@@ -294,4 +345,66 @@ angular.module('visits')
                 }).join("\r\n"));
             });
         };
+
+        self.getStatus = function (status) {
+            var statusText = "";
+
+            if(status === false){
+                statusText = "Canceled";
+            }
+            else{
+                statusText = "Not Canceled";
+            }
+
+            return statusText;
+        };
+
+        self.cancelVisit = function (id, visitStatus, visitPractitionerId, visitDate, visitDescription){
+            visitId = id;
+            var data = {};
+
+            if (visitStatus) {
+                data = {
+                    date: visitDate,
+                    description: visitDescription,
+                    practitionerId: visitPractitionerId,
+                    status: false
+                };
+            }else {
+                data = {
+                    date: visitDate,
+                    description: visitDescription,
+                    practitionerId: visitPractitionerId,
+                    status: true
+                };
+            }
+
+
+            url = "api/gateway/owners/*/pets/" + petId + "/visits/" + visitId;
+
+            $http.put(url, data).then(function () {
+                $http.get("api/gateway/visits/"+petId).then(function (resp) {
+                    self.visits = resp.data;
+                    self.sortFetchedVisits();
+                });
```

Comment:
We probably shouldn't be requesting all the visits again when we're just changing the status of a visit. Maybe request that single visit and then update the out of date visit in the list of visits that was already fetched to match it?